src/webapi/html_elements/slot.rs

```diff
@@ -0,0 +1,88 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::node::{INode, Node};
+use webapi::element::{IElement, Element};
+use webapi::html_element::{IHtmlElement, HtmlElement};
+use webapi::slotable::ISlotable;
+
+/// An enum which determines whether
+/// [SlotElement::assigned_nodes](struct.SlotElement.html#method.assigned_nodes) /
+/// [SlotElement::assigned_elements](struct.SlotElement.html#method.assigned_elements) will
+/// return the content assigned or its fallback content.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum SlotContentKind {
+    /// Will return the content assigned.
+    Default,
+    /// Will return the fallback content.
+    Fallback,
+}
+
+/// The HTML `<slot>` element represents a placeholder inside a web component that
+/// you can fill with your own markup, which lets you create separate DOM trees and
+/// present them together.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot)
+// https://html.spec.whatwg.org/multipage/scripting.html#htmlslotelement
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "HTMLSlotElement")]
+#[reference(subclass_of(EventTarget, Node, Element, HtmlElement))]
+pub struct SlotElement( Reference );
+
+impl IEventTarget for SlotElement {}
+impl INode for SlotElement {}
+impl IElement for SlotElement {}
+impl IHtmlElement for SlotElement {}
+impl ISlotable for SlotElement {}
+
+impl SlotElement {
+    /// The slot's name
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/name)
+    // https://html.spec.whatwg.org/multipage/scripting.html#attr-slot-name
+    #[inline]
+    pub fn name ( &self ) -> String {
+        js! (
+            return @{self}.name;
+        ).try_into().unwrap()
+    }
+
+    /// Setter of name.
+    #[inline]
+    pub fn set_name( &self, new_name: &str ) {
+        js! ( @(no_return)
+            @{self}.name = @{new_name};
+        );
+    }
+
+    /// Returns slot's assigned nodes.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedNodes)
+    // https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element:dom-slot-assignednodes
+    pub fn assigned_nodes( &self, kind: SlotContentKind ) -> Vec<Node> {
+        let is_flatten = match kind {
+            SlotContentKind::Default => false,
+            SlotContentKind::Fallback => true,
```

Comment:
Well, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedNodes), what `{ flatten: true }` actually means is that it will always return the fallback content of the `slot` defined in the `<slot></slot>` tag before the shadow tree is attached. And the **default behaviour** for `slot.assignedNode()` is `{ flatten: false }` which means the browsers will try to return any content assigned.

Reply:
Also, 

```rust
slot.assigned_node(SlotContentKind::Assigned);
```
looks kind of weird. 

So I changed it to
```rust
slot.assigned_node(SlotContentKind::Default);
```
.